src/integration-test/java/com/commercetools/sync/integration/commons/utils/CartDiscountITUtils.java

```diff
@@ -0,0 +1,150 @@
+
+    public static final CartDiscountTarget CART_DISCOUNT_TARGET_1 = LineItemsTarget.ofAll();
+    public static final CartDiscountTarget CART_DISCOUNT_TARGET_2 =
+        LineItemsTarget.of("sku = \"0123456789\" or sku = \"0246891213\"");
+
+    public static final ZonedDateTime JANUARY_FROM = ZonedDateTime.parse("2019-01-01T00:00:00.000Z");
+    public static final ZonedDateTime JANUARY_UNTIL = ZonedDateTime.parse("2019-01-31T00:00:00.000Z");
+    public static final ZonedDateTime FEBRUARY_FROM = ZonedDateTime.parse("2019-02-01T00:00:00.000Z");
+    public static final ZonedDateTime FEBRUARY_UNTIL = ZonedDateTime.parse("2019-02-28T00:00:00.000Z");
+
+    public static final String SORT_ORDER_1 = "0.1";
+    public static final String SORT_ORDER_2 = "0.2";
+
+    public static final CartDiscountDraft CART_DISCOUNT_DRAFT_1 =
+        CartDiscountDraftBuilder.of(LocalizedString.of(Locale.ENGLISH, CART_DISCOUNT_KEY_1),
+            CART_DISCOUNT_CART_PREDICATE_1,
+            CART_DISCOUNT_VALUE_1,
+            CART_DISCOUNT_TARGET_1,
+            SORT_ORDER_1,
+            false)
+                                .active(false)
+                                .description(CART_DISCOUNT_DESC_1)
+                                .validFrom(JANUARY_FROM)
+                                .validUntil(JANUARY_UNTIL)
+                                .build();
+
+    public static final CartDiscountDraft CART_DISCOUNT_DRAFT_2 =
+        CartDiscountDraftBuilder.of(LocalizedString.of(Locale.ENGLISH, CART_DISCOUNT_KEY_2),
+            CART_DISCOUNT_CART_PREDICATE_2,
+            CART_DISCOUNT_VALUE_2,
+            CART_DISCOUNT_TARGET_2,
+            SORT_ORDER_2,
+            false)
+                                .active(false)
+                                .description(CART_DISCOUNT_DESC_2)
+                                .validFrom(FEBRUARY_FROM)
+                                .validUntil(FEBRUARY_UNTIL)
+                                .build();
+
+
+    /**
+     * Deletes all cart discounts from CTP project, represented by provided {@code ctpClient}.
+     *
+     * @param ctpClient represents the CTP project the cart discounts will be deleted from.
+     */
+    public static void deleteCartDiscounts(@Nonnull final SphereClient ctpClient) {
+        queryAndExecute(ctpClient, CartDiscountQuery.of(), CartDiscountDeleteCommand::of);
+    }
+
+    /**
+     * Deletes all cart discounts from CTP projects defined by {@code CTP_SOURCE_CLIENT} and {@code CTP_TARGET_CLIENT}.
+     */
+    public static void deleteCartDiscountsFromTargetAndSource() {
+        deleteCartDiscounts(CTP_SOURCE_CLIENT);
+        deleteCartDiscounts(CTP_TARGET_CLIENT);
+    }
+
+
+    public static void populateSourceProject() {
+        CTP_SOURCE_CLIENT.execute(CartDiscountCreateCommand.of(CART_DISCOUNT_DRAFT_1)).toCompletableFuture().join();
+        CTP_SOURCE_CLIENT.execute(CartDiscountCreateCommand.of(CART_DISCOUNT_DRAFT_2)).toCompletableFuture().join();
```

Comment:
better do them in parallel to avoid the extra request -> longer time for tests to run.

Reply:
c8e1f48